src/lib/Libecl/ecl_verify.c

```diff
@@ -630,14 +630,49 @@ static struct ecl_attribute_def * ecl_find_attr_in_def(
 
 	if (attr_def) {
 		for (index = 0; index < limit; index++) {
-			if (!strcasecmp(attr_def[index].at_name, name))
-				return (&attr_def[index]);
+			char *pc = NULL;
+
+			if (strncasecmp(name, attr_def[index].at_name,
+					strlen(attr_def[index].at_name)) == 0) {
+				pc = name + strlen(attr_def[index].at_name);
+				if ((*pc == '\0') || (*pc == '.') || (*pc == ','))
+					return &(attr_def[index]);
+			}
 		}
 	}
 	return NULL;
 }
 
 /**
+ * @brief 	Return the type of attribute (public or read-only)
+ *
+ * @param[in]	attr_def	-	the attribute to check
+ *
+ * @return	int
+ * @retval	TYPE_ATTR_PUBLIC if the attribute is public
+ * @retval	TYPE_ATTR_READONLY otherwise
+ *
+ * @par Side Effects:
+ *	None
+ *
+ * @par MT-safe: Yes
+ */
+static int
+get_attr_type(struct ecl_attribute_def attr_def)
+{
+	/*
+	 * Consider an attr def public if it has any of the write flags set
+	 */
+	if (attr_def.at_flags & (ATR_DFLAG_USWR | ATR_DFLAG_OPWR | ATR_DFLAG_MGWR))
+		return TYPE_ATTR_PUBLIC;
+
+	else
+		return TYPE_ATTR_READONLY;
+
```

Comment:
SvWR is technically READ_ONLY to the users right? You wouldn't call it public, it is writable by the server/scheduler only. So, I personally think it's fair. There are only three classes: read only, public and all, is there a better class for SvWR ?